### ISSUE TITLE
Loading into PowerShell 5.1 (.Net Framework) environment fails with version 5.2 but works with 5.1.5

### ISSUE THREAD

**Issue by mithrandyr** [dotnet/SqlClient#2498]
### Describe the bug
When loading the Microsoft.Data.SqlClient assemblies into PowerShell 5.1, it fails on version 5.2 but works with version 5.1.5.
It fails with the error:
```
Exception message:  The type initializer for 'Microsoft.Data.SqlClient.SqlConnection' threw an exception.
Stack trace:
 System.TypeInitializationException: The type initializer for 'Microsoft.Data.SqlClient.SqlConnection' threw an exception. --->
                        System.TypeLoadException: Could not load type 'System.Runtime.Loader.AssemblyLoadContext' from assembly 'System.Runtime.Loader, Version=4.0.1.0,
                        Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
                           at Microsoft.Data.SqlClient.SqlDiagnosticListener..ctor(String name)
                           at Microsoft.Data.SqlClient.SqlConnection..cctor()
                           --- End of inner exception stack trace ---
                           at Microsoft.Data.SqlClient.SqlConnection..ctor()
                           at Microsoft.Data.SqlClient.SqlConnection..ctor(String connectionString)

```
**Here are the results from $PSVersionTable**
```
PSVersion                      5.1.19041.4291
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.4291
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### To reproduce
Include a complete code listing (or project/solution) that we can run to reproduce the issue.

Partial code listings, or multiple fragments of code, will slow down our response or cause us to push the issue back to you to provide code to reproduce the issue.

```PowerShell
#Launch Windows PowerShell (version 5.1, 64bit)
#Download the nuget package and extract the x64 version of the dlls (or create a sample project and compile against win-64)
#In the directory where the DLLs are, run this

#Will fail for 5.2; will succeed for 5.1.5
Add-Type -Path .\Microsoft.Data.SqlClient.dll
```

### Expected behavior
The DLL will load properly.

### Further technical details
Microsoft.Data.SqlClient version: 5.2
.NET target: .NET Standard (but fails on .NET framework 
SQL Server version: any
Operating system: Windows 10

**Comment by JRahnama** [comment]
@mithrandyr we will look into this and will get back to you.

**Comment by arellegue** [comment]
@mithrandyr I have investigated this issue  However, I was not able to reproduce the issue.

These are the steps taken to try to reproduce the issue:
1. Created a console app targeting .Net Framework 4.8.1 (since Powershell is using .Net Framework)
2. Added package MDS 5.1.5
3. Compiled the app.
4. Started powershell and changed directory to the bin\debug folder of the console app
5. Run the Add-Type -Path .\Microsoft.Data.SqlClient.dll command. 

The dll loaded without issue.

Repeat from step 2 above but reference MDS 5.2.0 instead. The dll loaded without issue as well.

This is the version of pwershell used:
Name                           Value                                                                                                                                                                                                                             
----                           -----                                                                                                                                                                                                                             
PSVersion                      5.1.22621.2506                                                                                                                                                                                                                    
PSEdition                      Desktop                                                                                                                                                                                                                           
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                                           
BuildVersion                   10.0.22621.2506                                                                                                                                                                                                                   
CLRVersion                     4.0.30319.42000                                                                                                                                                                                                                   
WSManStackVersion              3.0                                                                                                                                                                                                                               
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                                               
SerializationVersion           1.1.0.1
